tdrs-backend/tdpservice/users/signals.py

```diff
@@ -0,0 +1,30 @@
+"""Signals for the users app."""
+from django.db.models.signals import m2m_changed
+from django.dispatch import receiver
+from tdpservice.users.models import User
+from django.contrib.auth.models import Group
+from tdpservice.email.helpers.admin_notifications import email_system_owner_system_admin_role_change
+
+import logging
+logger = logging.getLogger()
+
+@receiver(m2m_changed, sender=User.groups.through)
+def user_group_changed(sender, instance, action, pk_set, **kwargs):
+    """Send an email to the System Owner when a user is assigned or removed from the System Admin role."""
+    ACTIONS = {
+            'PRE_REMOVE': 'pre_remove',
+            'PRE_ADD': 'pre_add',
+            'PRE_CLEAR': 'pre_clear'
```

Comment:
enum?

Reply:
there is no difference really between enum and having a dictionary, and I find dictionary more readable